iree/compiler/Conversion/LinalgToLLVM/KernelDispatch.cpp

```diff
@@ -69,189 +72,169 @@ static llvm::cl::opt<int> genericOpsWorkgroupTileSize(
-            workgroupTileSizes.push_back(genericOpsWorkgroupTileSize);
-          } else {
-            // Don't tile workgroup across reduction dimensions.
-            workgroupTileSizes.push_back(0);
-          }
-        }
-        return workgroupTileSizes;
-      }
-      // TODO(ataei): Set the parameters when we enable vectorization.
-      case TilingLevel::Level1Tiles:
-      case TilingLevel::Level2Tiles:
-        return {1, 1, 1};
-    }
+  if (contractionOp.isRowMajorBatchMatmul()) {
+    // TODO(ataei, ravishankarm): This should just use the configuration for
+    // matmul above. setting the tile size to 1 for all the batch dimensions.
+    TileSizesListType tileSizes = {
+        {1, batchMatmulWorkgroupTileSize, batchMatmulWorkgroupTileSize},
+        {1, batchMatmulL1TileSize, batchMatmulL1TileSize},
+        {1, batchMatmulL2TileSize, batchMatmulL2TileSize}};
+    SmallVector<int64_t, 4> nativeVectorSize = {1, batchMatmulL2TileSize,
+                                                batchMatmulL2TileSize};
+    IREE::HAL::LoweringConfig config =
+        getConfigAttr(tileSizes, nativeVectorSize, contractionOp->getContext());
+    setLoweringConfig(contractionOp, config);
+    return success();
   }
-
-  return {1, 1, 1};
+  return failure();
 }
 
-#define DEFINE_TILE_SIZE_FN(tilingLevel)                                      \
-  template <>                                                                 \
-  SmallVector<Value, 4> TileSizeFn::get<tilingLevel>(OpBuilder & builder,     \
-                                                     Operation * operation) { \
-    auto tileSizes = getTileSizes<tilingLevel>(operation);                    \
-    if (tileSizes.empty()) return {};                                         \
-    SmallVector<Value, 4> tileSizesVal;                                       \
-    tileSizesVal.reserve(tileSizes.size());                                   \
-    for (auto val : tileSizes) {                                              \
-      tileSizesVal.push_back(                                                 \
-          builder.create<ConstantIndexOp>(operation->getLoc(), val));         \
-    }                                                                         \
-    return tileSizesVal;                                                      \
+/// Sets the lowering configuration for dispatch region with root op being a
+/// generic op.
+LogicalResult setRootConfig(linalg::GenericOp genericOp) {
+  int64_t numOuterParallelLoops = getNumOuterParallelLoops(genericOp);
+  SmallVector<int64_t, 4> workgroupTileSizes(numOuterParallelLoops,
+                                             genericOpsWorkgroupTileSize);
+  SmallVector<int64_t, 4> innerTileSizes(numOuterParallelLoops, 1);
+  // At the HAL level only the inner 3 parallel loops are tiled. Set the tile
+  // size as 0 for all the outer dimensions greater than 3.
+  for (int64_t i = 0; i < numOuterParallelLoops - 3; i++) {
```

Comment:
We actually need to tile the 3 outer loops otherwise tiles will be strided and not contiguous

Reply:
Could you clarify that. THis is required for correctness. The dispatch region creation follows this when it does tile + distribute. Of the outer parallel loops, upto the inner 3 will be tiled. If we dont do this here, it will be incosistent.

Reply:
Ah Yes this loop is setting inner parallel dims to 0, 
I was confused by the comment above got inner and outter swapped, it should be
```c++
// At the HAL level only the outter 3 parallel loops are tiled. 
// Set the tile size as 0 for all the inner dimensions greater than 3.
```

Reply:
Ok, ill reword this.